src/Sign.Cli/AzureAuthOptions.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE.txt file in the project root for more information.
+
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using Azure.Core;
+using Azure.Identity;
+using Sign.Core;
+
+namespace Sign.Cli
+{
+    internal sealed class AzureAuthOptions
+    {
+        internal Option<bool> ManagedIdentityOption { get; } = new(["-azm", "--azure-managed-identity"], getDefaultValue: () => false, Resources.ManagedIdentityOptionDescription);
+        internal Option<string?> TenantIdOption { get; } = new(["-azt", "--azure-tenant-id"], Resources.TenantIdOptionDescription);
+        internal Option<string?> ClientIdOption { get; } = new(["-azi", "--azure-client-id"], Resources.ClientIdOptionDescription);
+        internal Option<string?> ClientSecretOption { get; } = new(["-azs", "--azure-client-secret"], Resources.ClientSecretOptionDescription);
+
+        internal void AddOptionsToCommand(Command command)
+        {
+            command.AddOption(ManagedIdentityOption);
+            command.AddOption(TenantIdOption);
+            command.AddOption(ClientIdOption);
+            command.AddOption(ClientSecretOption);
+        }
+
+        internal TokenCredential? CreateTokenCredential(InvocationContext context)
+        {
+            bool useManagedIdentity = context.ParseResult.GetValueForOption(ManagedIdentityOption);
+
+            if (useManagedIdentity)
+            {
+                return new DefaultAzureCredential();
```

Comment:
The `--azure-managed-identity` is not exactly an accurate option name. `DefaultAzureCredential` has many behaviors, including token acquisition via the Az CLI. 
https://learn.microsoft.com/en-us/dotnet/api/azure.identity.defaultazurecredential?view=azure-dotnet

That being said, I think using `DefaultAzureCredential` unlocks a ton of capabilities. For example, GitHub Actions OIDC + the azure/login action light up Az CLI meaning `DefaultAzureCredential` can hook into those ambient credentials, much like a managed identity.

Another scenario which is not clear is user-assigned managed identities. Generally you need to specify the client ID somehow to use a user-assigned managed identity.

My suggestion is to change the option name to `--azure-default-token-credential` and introduce another `--azure-managed-identity` which an option value which is the client ID. This makes the `ManagedIdentityCredential` get used, with the optional `clientId` ctor parameter specified.

Reply:
You could also have an `--azure-interactive` switch which enables the interactive mode on `DefaultAzureCredential`
https://learn.microsoft.com/en-us/dotnet/api/azure.identity.defaultazurecredential.-ctor?view=azure-dotnet#azure-identity-defaultazurecredential-ctor(system-boolean)

Reply:
See my comment earlier in a response to what @clairernovotny was asking about me moving this out to a separate class.

Reply:
I will copy this to the other thread.